### PR TITLE
fixing format for principal identity to be domain/sub

### PR DESCRIPTION
**Pull request by akoserwal**
## Describe your changes

Updating the principal identity format from domain: sub to "domain (without ".com"/sub)

- ...

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: .inventory-api.yaml [hunk 1]
```diff
@@ -6,6 +6,11 @@ server:
     address: localhost:9000
 authn:
    allow-unauthenticated: true
+#   oidc:
+#     authn-server-url: http://localhost:8084/realms/redhat-external
+#     skip-client-id-check: true
+#     skip-issuer-check: true
+#     principal-user-domain: localhost
```
Comment (josejulio) [inline .inventory-api.yaml:13]: Is this comited on purpose? should we remove if it is no longer relevant ?

File: internal/authn/oidc/oidc.go [hunk 2]
```diff
@@ -97,23 +96,27 @@ func issuerCheck(issuer string, domain string) bool {
...
+		issuer = issuer[:dotIndex]
+	}
+
+	if strings.HasSuffix(issuer, "."+domain) {
 		return true
 	}
-	return false
+
+	return issuer == domain
 }
```
Comment (josejulio) [inline internal/authn/oidc/oidc.go:122]: Could we add some tests against this issuerCheck?

File: internal/authn/oidc/oidc.go [hunk 3]
```diff
@@ -97,23 +96,27 @@ func issuerCheck(issuer string, domain string) bool {
...
+	if dotIndex := strings.LastIndex(issuer, "."); dotIndex != -1 {
+		issuer = issuer[:dotIndex]
+	}
+
+	if strings.HasSuffix(issuer, "."+domain) {
 		return true
 	}
-	return false
+
+	return issuer == domain
```
Comment (alechenninger) [inline internal/authn/oidc/oidc.go:121]: So the domain is not _literally_ the DNS domain of the issuer – it is just an arbitrary/abstract "domain" identifier (as in, "trust domain"). So what I would do is when configuring an issuer, have the "domain" that is used be configurable per that issuer. You have the "principal-user-domain" config already – basically use that as long as the token validates with a key from that issuer.

File: .inventory-api.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (akoserwal) [reply .inventory-api.yaml:13]: This file used for the default run option for inventory API locally,  where I added oidc config comment as an example. If someone wants to test with sso integration locally

Comment (akoserwal) [reply .inventory-api.yaml:13]: Yes, I add on purpose. For devs who want to run a local keycloak and test oidc flow

File: internal/authn/oidc/oidc.go (hunk 2 above)
Comment (akoserwal) [reply internal/authn/oidc/oidc.go]: remove the issuer check

File: internal/authn/oidc/oidc.go (hunk 3 above)
Comment (akoserwal) [reply internal/authn/oidc/oidc.go]: fixed

**Review by josejulio — APPROVED** [review]

**Pull request merged**